smarts/core/chassis.py

```diff
@@ -629,20 +651,30 @@ def control(self, throttle=0, brake=0, steering=0):
         assert 0 <= brake <= 1, f"brake ({brake}) must be in [0, 1]"
         assert -1 <= steering <= 1, f"steering ({steering}) must be in [-1, 1]"
 
+        # If we apply brake at low speed using reverse torque
+        # the vehicle starts to roll back. we need to apply a condition
+        # on brake such that, the reverse torque is only applied after
+        # a threshold is passed for vehicle velocity.
+        # Thus, brake is applied if: vehicle speed > 1/36 (m/s)
+        if brake > 0 and self.longitudinal_lateral_speed[0] < 1 / 36:
```

Comment:
Interestingly, probably the most important optimization in this entire PR was the addition of the `brake > 0` check above. 

This prevents the call to `longitudinal_lateral_speed` (which makes a pybullet API call) -- as well as the subsequent call to `pose` -- in the majority of cases.  (Vehicles rarely hit the brakes, lol!)  Because `control()` is called for every pybullet _substep_, this is a _huge_ win.